pymilvus/stage/file_utils.py

```diff
@@ -0,0 +1,42 @@
+import os
+from pathlib import Path
+from typing import Any, List, Tuple, Union
+
+
+class FileUtils:
+    @staticmethod
+    def process_local_path(
+        local_path: str,
+    ) -> Union[tuple[list[Path], int], tuple[list[str], int], tuple[list[Any], int]]:
+        """
+        Get all file paths under the local_path, recursively.
+        If local_path is a file, return it directly.
+        Returns a tuple: (list of file paths, total size in bytes)
+        """
+        if not Path(local_path).exists():
+            error_message = f"Path does not exist: {local_path}"
+            raise ValueError(error_message)
+
+        path = Path(local_path)
+        if path.is_file():
+            return [path.resolve()], path.stat().st_size
+        if path.is_dir():
+            return FileUtils.find_files_recursively(local_path)
+
+        return [], 0
+
+    @staticmethod
+    def find_files_recursively(folder_path: str) -> Tuple[List[str], int]:
+        """
+        Recursively finds all files under a folder and calculates total size.
+        """
+        result = []
+        total_size = 0
+
+        for root, _, files in os.walk(folder_path):
+            for file in files:
+                full_path = Path(root) / file
+                result.append(str(full_path.resolve()))
+                total_size += full_path.stat().st_size
+
+        return result, total_size
```

Comment:
It's recommanded to take full advantage of pathlib to support windows platform. There're two things to be cared of:
1. local path should always use pathlib.Path instead of str
2. Remote path(endpoint or url) should always use PosixPath.

For example In your Code `Path(root) / file` only fits in posix-like system. Windows users are going to complain about it.

If you don't intend to support windows platform, Please doc or comment it properly.